webview/src/plots/components/ZoomedInPlot.tsx

```diff
@@ -23,8 +29,31 @@ export const ZoomedInPlot: React.FC<ZoomedInPlotProps> = ({
     }
   }, [])
 
+  const onPlotChange = () => {
```

Comment:
Tried to write a jest test for this custom action but it was failing for some reason:

```
  it('should add a custom action in zoomed in plot modal for exporting raw data', async () => {
    renderAppWithOptionalData({
      template: complexTemplatePlotsFixture
    })

    expect(screen.queryByTestId('modal')).not.toBeInTheDocument()

    const plot = within(screen.getAllByTestId(/^plot_/)[0]).getByRole('button')

    fireEvent.click(plot)

    const modal = screen.getByTestId('modal')

    expect(screen.getByTestId('modal')).toBeInTheDocument()

    await waitFor(
      () =>
        // eslint-disable-next-line testing-library/no-node-access
        expect(modal.querySelectorAll('.marks')[0]).toBeInTheDocument(),
      { timeout: 5000 }
    )

    const actionsMenu = within(modal).getByTitle('Click to view actions')

    expect(actionsMenu).toBeInTheDocument()

    fireEvent.click(actionsMenu)

    expect(screen.getByText('Save as Raw Data')).toBeInTheDocument() // fails 
  })
```

Debugging showed that the append seemed to be running but the HTML of the vega actions menu only showed two actions instead of three 🤔

Reply:
> 92.4%

Looked over the code coverage, I believe the missing percentage is due the new lines in `ZoomedInPlot`. Any ideas why the test above isn't working or an alternative way to test for the custom action?

Reply:
> try adding a waitFor around expect(screen.getByText('Save as Raw Data')).toBeInTheDocument() to see if there is a timing issue there

Tried that as well with an extended timeout but I just hit a timeout error :(

Reply:
Figured out the issue! Look me way too long to realize it 🤦‍♀️ but the code wasn't actually running correctly due to us mocking the MutationObserver in `setup-tests.js`. I think we're safe to delete the mocked function since it was added in #1365 for the use of the api that no longer exists(https://github.com/iterative/vscode-dvc/blob/f026ac15f0d8d6cae7263132e66abee6b9246d0c/webview/src/shared/components/theme/Theme.tsx#L3).  

Will delete the mock for now!